src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -48,17 +51,58 @@ public ArmSubsystem(MoInput input) {
         armChooser.addOption("Direct (PID)", pidArmControlCommand);
 
         MoShuffleboard.getInstance().matchTab.add("Arm Control Mode", armChooser).withWidget(BuiltInWidgets.kComboBoxChooser);
+ 
+        this.zero();
+        MoShuffleboard.getInstance().settingsTab
+            .add("Recalculate Arm Position", new InstantCommand(this::zero));
     }
 
     @Override
     public void periodic() {
         this.updateDefaultCommand();
     }
 
+    public void zero() {
+        this.leftShoulder.getEncoder().setPosition(
+            ((this.leftShoulder.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle)
+                .getPosition() - MoPrefs.absShoulderZero.get() + 1) % 1) * MoPrefs.shoulderAbsToRel.get()
+        );
+        this.wrist.getEncoder().setPosition(
+            ((this.wrist.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle)
+                .getPosition() - MoPrefs.absWristZero.get() + 1) % 1) * MoPrefs.wristAbsToRel.get()
+        );
+    }
+
+    private boolean shoulderMoveLimited(double movement) {
+        double pos = this.leftShoulder.getEncoder().getPosition();
+        if (movement > 0) {
+            return pos >= MoPrefs.relShoulderMax.get();
+        }
+        if (movement < 0) {
+            return pos <= 0;
+        }
+        return false;
+    }
+
+    private boolean wristMoveLimited(double movement) {
+        double pos = this.wrist.getEncoder().getPosition();
+        if (movement > 0) {
+            return pos >= MoPrefs.relWristMax.get();
+        }
+        if (movement < 0) {
+            return pos <= 0;
+        }
+        return false;
+    }
+
     /**
      * @param target -1.0 to 1.0, each being the max speed in that direction
      */
     public void adjustShoulders(boolean pid, double target) {
+        if (shoulderMoveLimited(target)) {
+            leftShoulder.stopMotor();
+        }
```

Comment:
If we overextend the limit, we don't want to completely stop the motor. We still want the motor to be able to drive away from the limit, back into the allowed range.